### PR TITLE
Refactoring `ConvModule` by removing `act_cfg`

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:

- [x] Replace `act_cfg` with `activation_callable`
- [x] Remove `build_activation_layer`

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — DISMISSED** [review]
LGTM!

File: src/otx/algo/action_classification/backbones/x3d.py [hunk 1]
```diff
@@ -139,13 +139,13 @@ def __init__(
...
             norm_cfg=self.norm_cfg,
-            act_cfg=None,
+            activation_callable=None,
         )
 
         if self.se_ratio is not None:
             self.se_module = SEModule(planes, self.se_ratio)
 
-        self.relu = build_activation_layer(self.act_cfg) if self.act_cfg else build_activation_layer({})
+        self.relu = self.activation_callable() if self.activation_callable else nn.ReLU(inplace=True)
```
Comment (eunwoosh) [inline src/otx/algo/action_classification/backbones/x3d.py:148]: If we can set other activation layer, then how about changing name from `self.relu`?

File: src/otx/algo/classification/backbones/efficientnet.py [hunk 2]
```diff
@@ -212,7 +214,7 @@ def __init__(
             )
         else:
             self.fc2 = nn.Linear(in_features=mid_channels, out_features=channels)
-        self.sigmoid = build_activation_layer({"type": out_activation})
+        self.sigmoid = out_activation_callable()
```
Comment (eunwoosh) [inline src/otx/algo/classification/backbones/efficientnet.py:217]: Same as above.

File: src/otx/algo/detection/backbones/csp_darknet.py [hunk 3]
```diff
@@ -148,7 +149,7 @@ def __init__(
         arch_ovewrite: list | None = None,
         spp_kernal_sizes: tuple[int, ...] = (5, 9, 13),
         norm_cfg: dict | None = None,
-        act_cfg: dict | None = None,
+        activation_callable: Callable[..., nn.Module] | None = partial(nn.LeakyReLU, negative_slope=0.1),
```
Comment (eunwoosh) [inline src/otx/algo/detection/backbones/csp_darknet.py]: It seems that Swish is used as default, but it's changed to leakyrelu now. is it ok?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work! I left some comments. Please take a look.

File: src/otx/algo/action_classification/backbones/x3d.py (hunk 1 above)
Comment (sungchul1) [reply src/otx/algo/action_classification/backbones/x3d.py:148]: I think it depends on pretrained weights, if changing this we must add changing keys of pretrained weights as well.

File: src/otx/algo/detection/backbones/csp_darknet.py (hunk 3 above)
Comment (sungchul1) [reply src/otx/algo/detection/backbones/csp_darknet.py]: Thanks for finding, updated at 9f0d5a7.

**Review by eunwoosh — APPROVED** [review]

**Comment by harimkang** [comment]
I've canceled the action of this PR due to issues with the integration test at the moment, please update when this PR is merged. https://github.com/openvinotoolkit/training_extensions/pull/3815

**Review by kprokofi — COMMENTED** [review]
Thank you for the update. I have a general question. 
Can't we pass and use already instantiated activation instead of activation class (partial)?

**Comment by sungchul1** [comment]
> Thank you for the update. I have a general question. Can't we pass and use already instantiated activation instead of activation class (partial)?

I considered this before, but I haven't proceeded for now because original `ConvModule`'s purpose is an instantiation using given dict config, such as `conv_cfg`, `norm_cfg`, or `act_cfg`. 
I just updated this module maintaining this purpose, but I will try your suggestion if you or others (@harimkang, @eunwoosh, or any others 😊) think that using conv, norm, or activation instances as is is needed.
I think adding this is not hard work.

**Comment by eunwoosh** [comment]
> > Thank you for the update. I have a general question. Can't we pass and use already instantiated activation instead of activation class (partial)?
> 
> I considered this before, but I haven't proceeded for now because original `ConvModule`'s purpose is an instantiation using given dict config, such as `conv_cfg`, `norm_cfg`, or `act_cfg`. I just updated this module maintaining this purpose, but I will try your suggestion if you or others (@harimkang, @eunwoosh, or any others 😊) think that using conv, norm, or activation instances as is is needed. I think adding this is not hard work.

I actually think this approach when reviewing norm refactoring but it also can be activation layer. I think current approach is ok but Kirill's opinion is also good.

**Comment by sungchul1** [comment]
> Thank you for the update. I have a general question. Can't we pass and use already instantiated activation instead of activation class (partial)?

I'll try this targeting the next version as a new minor feature, conv and normalization can be also appiled.

**Review by harimkang — APPROVED** [review]
